src/client/api/pythonApi.ts

```
@@ -57,9 +57,11 @@ export class PythonApiProvider implements IPythonApiProvider {
         @inject(IPythonExtensionChecker) private extensionChecker: IPythonExtensionChecker
     ) {}
 
-    public getApi(): Promise<PythonApi> {
-        this.init().catch(noop);
-        return this.api.promise;
+    public async getApi(): Promise<PythonApi | undefined> {
+        if (await this.init().catch(noop)) {
```

Why are we swallowing the exceptions?